File: Java/WorkManagerDemo/app/src/main/java/com/example/workmanagerdemo/MainActivity.java
package com.example.workmanagerdemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.work.Data;
import androidx.work.OneTimeWorkRequest;
import androidx.work.WorkInfo;
import androidx.work.WorkManager;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.example.workmanagerdemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    public static final String COUNT_LIMIT_KEY = "count_limit_key";
    public static final String STATUS_KEY = "status_key";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

//        Constraints constraints = new Constraints.Builder()
//                .setRequiresCharging(true)
//                .build();

        Data data = new Data.Builder()
                .putInt(COUNT_LIMIT_KEY, 100)
                .build();

        final OneTimeWorkRequest oneTimeWorkRequest = new OneTimeWorkRequest.Builder(DemoWorker.class)
//                .setConstraints(constraints)
                .setInputData(data)
                .build();

        binding.startButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                WorkManager.getInstance(getApplicationContext()).enqueue(oneTimeWorkRequest);
            }
        });

        WorkManager.getInstance(getApplicationContext())
                .getWorkInfoByIdLiveData(oneTimeWorkRequest.getId())
                .observe(this, new Observer<WorkInfo>() {
                    @Override
                    public void onChanged(WorkInfo workInfo) {
                        if (workInfo != null) {
                            binding.tvStatus.setText(workInfo.getState().name());

                            if (workInfo.getState().isFinished()) {
                                Data outData = workInfo.getOutputData();
                                String message = outData.getString(STATUS_KEY);
                                Toast.makeText(getApplicationContext(), message, Toast.LENGTH_LONG).show();
                            }
                        }
                    }
                });

    }
}

File: Kotlin/LiveDataDemo/app/src/main/java/com/example/livedatademo/MainActivity.kt
package com.example.livedatademo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.livedatademo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private lateinit var viewModel: MainActivityViewModel
    private lateinit var viewModelFactory: MainActivityViewModelFactory

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        viewModelFactory = MainActivityViewModelFactory(23)
        viewModel = ViewModelProvider(this, viewModelFactory).get(MainActivityViewModel::class.java)

        viewModel.totalData.observe(this, Observer {
            binding.countTextView.text = it.toString()
        })

        binding.countButton.setOnClickListener {
            viewModel.setTotal(binding.instertText.text.toString().toInt())
        }
    }
}
